trivial/flake.nix

```diff
@@ -1,6 +1,10 @@
 {
   description = "A very basic flake";
 
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
```

Comment:
By default It seems that the registry for `nixpkgs` uses `nixpkgs-unstable`.

Changing the default to `nixos-unstable` is pertinent, but need- to be discussed separately.

Reply:
thanks for raising that point. I see this as yet another problem: flakes can be used to define systems, potentially used with nixos-rebuild. Thus defaulting to nixpkgs-unstable that doesn't run the bootloading tests as nixos-unstable does (I think ?) is a bad default, potentially making for a real bad experience. The channels are usually a few days apart with nixos-unstable even being fresher than nixpkgs-unstable at times (dunno how, but I remember seeing it once). 
I will leave it like that for now and will do the change in another PR if requested.

Reply:
Would it make sense for this one to use nixpkgs-unstable and there to be a separate template specifically for setting up a nixos template specifically for flakes defining systems?

Reply:
We could have templates for every usecase but that might be overwhelming. What we could do instead is to do like for the generated configuration.nix, i.e., add a comment just on top `use "?ref=nixpkgs-unstable" to get more recent but less tested packages"`. Since the 2 channels are usually a few days apart, I dont see this as a huge win but if people think it is, looks like a good compromise

Reply:
I think that nixos-unstable is the right choice pretty much always, because NixOS tests do test the NixOS modules but they also test that the software itself isn't broken, which you do care about on non NixOS systems.